l10n_de_tax_statement/tests/test_l10n_de_tax_statement.py

```diff
@@ -0,0 +1,354 @@
+
+        with self.assertRaises(UserError):
+            self.statement_1.write({'name': 'Test Name Modified'})
+        with self.assertRaises(UserError):
+            self.statement_1.write({'state': 'posted'})
+        with self.assertRaises(UserError):
+            self.statement_1.write({'date_posted': fields.Datetime.now()})
+        with self.assertRaises(UserError):
+            self.statement_1.unlink()
+        for line in self.statement_1.line_ids:
+            self.assertTrue(line.view_base_lines())
+            self.assertTrue(line.view_tax_lines())
+            with self.assertRaises(UserError):
+                line.unlink()
+
+        self.assertEqual(self.statement_1.tax_total, 0.)
+
+    def test_03_reset(self):
+        self.statement_1.reset()
+        self.assertEqual(self.statement_1.state, 'draft')
+        self.assertFalse(self.statement_1.date_posted)
+
+        self.assertEqual(self.statement_1.tax_total, 0.)
+
+        self.statement_1.statement_update()
+        for line in self.statement_1.line_ids:
+            self.assertTrue(line.view_base_lines())
+            self.assertTrue(line.view_tax_lines())
+
+    def test_04_write(self):
+        self.statement_1.post()
+        with self.assertRaises(UserError):
+            self.statement_1.write({'name': 'Test Name'})
+
+        self.assertEqual(self.statement_1.tax_total, 0.)
+
+    def test_05_unlink_exception(self):
+        self.statement_1.post()
+        with self.assertRaises(UserError):
+            self.statement_1.unlink()
+
+    def test_06_unlink_working(self):
+        self.statement_1.unlink()
+
+    def test_07_update_exception1(self):
+        self.statement_1.post()
+        with self.assertRaises(UserError):
+            self.statement_1.statement_update()
+
+    def test_08_update_exception2(self):
+        self.config.unlink()
+        with self.assertRaises(UserError):
+            self.statement_1.statement_update()
+
+    def test_09_update_working(self):
+        self.invoice_1._onchange_invoice_line_ids()
+        self.invoice_1.action_invoice_open()
+        self.statement_1.statement_update()
+        self.assertEqual(len(self.statement_1.line_ids.ids), 47)
+
+        _25 = self.statement_1.line_ids.filtered(lambda r: r.code == '25')
+        _26 = self.statement_1.line_ids.filtered(lambda r: r.code == '26')
+        _27 = self.statement_1.line_ids.filtered(lambda r: r.code == '27')
+
+        self.assertEqual(len(_25), 1)
+        self.assertEqual(len(_26), 1)
+        self.assertEqual(len(_27), 1)
+
+        self.assertFalse(_25.format_base)
+        self.assertFalse(_25.format_tax)
+        self.assertTrue(_25.is_group)
+        self.assertTrue(_25.is_readonly)
+
+        self.assertEqual(_26.format_base, '100.00')
+        self.assertFalse(_26.is_group)
+        self.assertTrue(_26.is_readonly)
+
+        self.assertEqual(_27.format_base, '50.00')
+        self.assertFalse(_27.is_group)
+        self.assertTrue(_27.is_readonly)
```

Comment:
```suggestion
        self.assertFalse(_27.is_readonly)
```

Reply:
I have applied your proposed change. Thanks a lot for your intervention!